zktraffic/endpoints/stats_server.py

```diff
@@ -77,30 +79,53 @@ def wakeup(self):
   def has_stats(self):
     return len(self._get_stats('per_path')) > 0
 
-  def _get_stats(self, name, prefix=''):
+  def _get_stats(self, name, prefix='', output_array=False):
     stats_by_opname = self._stats.stats(name, self._max_results)
 
-    stats = {}
+    if output_array:
+        stats_arr = []
+    else:
+        stats = {}
     for opname, opstats in stats_by_opname.items():
       for path, value in opstats.items():
-        stats["%s%s%s" % (prefix, opname, path)] = value
-
-    return stats
+        if output_array:
+            #TODO: split prefix to each own key
+            #TODO: when per_ip: split path by ':' and add IPs to their own key
+            if prefix.endswith('/'):
+                prefix=prefix[:-1]
+            tmp_dict={"opname": opname, "path": "%s%s" % (prefix, path), "value":value}
+            stats_arr.append(tmp_dict)
+        else:
+            stats["%s%s%s" % (prefix, opname, path)] = value
+    if output_array:
+        stats_json=json.dumps(stats_arr)
+        return stats_json
+    else:
+        return stats
 
   @HttpServer.route("/json/paths")
-  def json_paths(self):
-    return self._get_stats('per_path')
+  def json_paths(self, array_output=False):
+    array_get = HttpServer.request.GET.get("array_out")
+    if array_get and array_get.lower() in ['true', '1', 't', 'y', 'yes']:
+        array_output=True
+    return self._get_stats('per_path','',array_output)
 
   @HttpServer.route("/json/ips")
-  def json_ips(self):
-    return self._get_stats('per_ip', 'per_ip/')
+  def json_ips(self, array_output=False):
+    array_get = HttpServer.request.GET.get("array_out")
+    if array_get and array_get.lower() in ['true', '1', 't', 'y', 'yes']:
+        array_output=True
+    return self._get_stats('per_ip', 'per_ip/', array_output)
 
   @HttpServer.route("/json/auths")
-  def json_auths(self):
-    return self._get_stats('per_auth', 'per_auth/')
+  def json_auths(self, array_output=False):
+    array_get = HttpServer.request.GET.get("array_out")
+    if array_get and array_get.lower() in ['true', '1', 't', 'y', 'yes']:
+        array_output=True
+    return self._get_stats('per_auth', 'per_auth/', array_output)
 
   @HttpServer.route("/json/auths-dump")
-  def json_auths_dump(self):
+  def json_auths_dump(self, array_output=False):
```

Comment:
Nothing seem to be consuming this parameter in the function body, might not be needed.

Reply:
correct, that was leftover from a test. I removed it.